src/cpp/core/file_lock/AdvisoryFileLock.cpp

```diff
@@ -0,0 +1,179 @@
+/*
```

Comment:
We can't see it from the diff due to the new `FileLock.cpp`, but I'm assuming that `AdvisoryFileLock.cpp` is essentially identical to the previous `FileLock.cpp`. Is that correct? (if there are no changes I'm thinking then no review is necessary)